### PR TITLE
Bug26294

### PR DESCRIPTION
**Pull request by asn-d6**

### CODE REVIEW

File: src/feature/hs/hs_service.c [hunk 1]
```diff
@@ -2289,19 +2289,12 @@ update_all_descriptors_intro_points(time_t now)
 /* Return true iff the given intro point has expired that is it has been used
  * for too long or we've reached our max seen INTRODUCE2 cell. */
 STATIC int
-intro_point_should_expire(const hs_service_intro_point_t *ip,
+intro_point_should_expire(hs_service_intro_point_t *ip,
```
Comment (teor2345) [inline src/feature/hs/hs_service.c]: You could probably leave the const in here, if you want.

Comment (asn-d6) [reply src/feature/hs/hs_service.c]: Ouch that was left accidentally in from previous experimentations. Good catch.

Comment (asn-d6) [reply src/feature/hs/hs_service.c]: Fixed in 45f590055e67d122cc3e6cf88a88418fe80ddb0d.

File: src/feature/hs/hs_service.c [hunk 2]
```diff
@@ -2514,6 +2503,33 @@ rotate_all_descriptors(time_t now)
...
 }
 
+/** Check the replay cache of this intro point, and wipe it if needed. */
+STATIC void
+check_intro_point_replay_cache(hs_service_intro_point_t *ip)
+{
+  if ((size_t) replaycache_size(ip->replay_cache) >=
+      ip->replay_cache_max_introduce2) {
+    replaycache_free(ip->replay_cache);
+    ip->replay_cache = replaycache_new(0, 0);
```
[the remark this answers is not in the document]
Comment (asn-d6) [reply src/feature/hs/hs_service.c]: Fixed in 17905bac4384ea9d156425f4b11435c4def4b40f.

File: src/feature/hs/hs_service.c [hunk 3]
```diff
@@ -2514,6 +2503,33 @@ rotate_all_descriptors(time_t now)
   } FOR_EACH_SERVICE_END;
 }
 
+/** Check the replay cache of this intro point, and wipe it if needed. */
+STATIC void
+check_intro_point_replay_cache(hs_service_intro_point_t *ip)
+{
+  if ((size_t) replaycache_size(ip->replay_cache) >=
+      ip->replay_cache_max_introduce2) {
```
[the remark this answers is not in the document]
Comment (asn-d6) [reply src/feature/hs/hs_service.c]: appveyor was warning about this in travis that's why it's there. :/

**Pull request closed without merging**